include/fieldperp.hxx

```diff
@@ -59,8 +59,9 @@ class FieldPerp : public Field {
    * will be shared (non unique)
    */
   FieldPerp(const FieldPerp& f)
-      : Field(f.fieldmesh), yindex(f.yindex), nx(f.nx), nz(f.nz), data(f.data),
-        location(f.location) {}
+      : Field(f.fieldmesh), yindex(f.yindex), nx(f.nx), nz(f.nz), data(f.data) {
+    location = f.location;
```

Comment:
Might be worth adding a constructor for `Field` that takes the location and then passing `f.location` to that

Reply:
That's taken care of in #1624. In fact this update is implemented independently in #1624. Might be better just to close this one?